src/Filter/Shape/Rename.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Membrane\Filter\Shape;
+
+use Membrane\Filter;
+use Membrane\Result\Message;
+use Membrane\Result\MessageSet;
+use Membrane\Result\Result;
+
+class Rename implements Filter
+{
+    public function __construct(
+        private readonly string $old,
+        private readonly string $new
+    )
+    {
```

Comment:
Wonder if its worth throwing an exception if $old === $new. In this instance the filter would behave as a delete filter, which is almost certainly a developer error

Reply:
Rename: I've added a RuntimeException if `$old === $new`

Pluck: I've changed it to `$from = $value[$this->fieldSet] ?? [];` 
    This still passes all tests since an empty array is considered a list so `if (is_array($from) && !array_is_list($from))` will return false.